Solution1/calculator_app/Program.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+
+// Приложение последовательно предлагает пользователю ввести первый и второй операнды,
+// а также символ необходимой операции. Вычислить значение выражения согласно требуемой
+// операции и вывести результат на экран. Поддерживаемые операции:
+// + - / * ^ (возведение в степень) 
+
+namespace CalculatorApp
+{
+    class Program
+    {
+        static bool TryToConvertInsertedOperandToInteger(string insertedOperand)
+        {
+            double integerOperand;
+            bool operandIsInteger = double.TryParse(insertedOperand, out integerOperand);
+            return operandIsInteger;
+        }
+
+        static double ConvertInsertedOperandToInteger(string insertedOperand)
+        {
+            double integerOperand;
+            bool operandIsInteger = double.TryParse(insertedOperand, out integerOperand);
+            return integerOperand;
+        }
+
+        static bool CheckIsOperationPossible(string insertedOperation)
+        {
+            bool isOperationPossible;
+            if (insertedOperation == "+" || insertedOperation == "-" || insertedOperation == "*" || insertedOperation == "/" || insertedOperation == "^")
+            {
+                isOperationPossible = true;
+            }
+            else
+            {
+                isOperationPossible = false;
+            }
+            return isOperationPossible;
+        }
+
+        static void Main()
+        {
+            do
+            {
+                bool isOperandInteger;
+                double firstOperand;
+                do
+                {
+                    Console.WriteLine("Enter first integer operand");
+                    string insertedFirstOperand = Console.ReadLine();
+
+                    isOperandInteger = TryToConvertInsertedOperandToInteger(insertedFirstOperand);
```

Comment:
So hard to understand what's going on
Why you shouldn't just use our check/parse functions from first lesson? (Amended with asked fixes please)
And there woudln't be a need in these two strange functions. Just ask user for a number until he will input correct symbol

Reply:
Done